### PR TITLE
Fix inline images within wide/full Image block captions shown with full width

### LINKED ISSUE #22808: Image block: Display bugs with inline images in captions
**Describe the bug**
If someone puts LaTex in an image caption, it renders as an image and adopts that block's alignment. Also, the LaTex does not render inline as expected, and appears as a separate block.

**To reproduce**
1. Add an image and add some LaTeX to its caption, like `$latex d_f$`
2. Set image to Wide or Full width
3. Add another image with LaTex in its caption. Keep default alignment.

**Expected behavior**
LaTex renders as a small image, and it should appear inline like this:
Screen Shot 2020-06-01 at 1 27 31 PM

Instead, it's rendering like this:

**Screenshots**
Screen Shot 2020-06-01 at 1 18 32 PM

**Editor version (please complete the following information):**
- WordPress version: 5.4.1
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Has Gutenberg
- If the Gutenberg plugin is installed, which version is it? 8.1.0

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser: Chrome
 - Version: Current on both

**Additional context**
Brought up here: https://wordpress.com/forums/topic/latex-in-caption-of-full-width-image/

This CSS fixes the too-wide LaTeX and keeps it inline:
```
.entry .entry-content .wp-block-image img :not(.latex) {
    display: block;
}
.wp-block-image.alignfull img :not(.latex), .wp-block-image.alignwide img :not(.latex) {
    width: 100%;
}
```

### PR DESCRIPTION
**Pull request by oxyc**
## Description

Fixes #22808 by not letting alingwide and alignfull `width: 100%` css style target inline images within captions. The use case of the issue is LaTeX which is added through JetPack but there could potentially be other plugins doing something similar so I'd say it's good to tighten this selector. The specificity of the selector remains the same even with the `>`-selector.

Gutenberg also supports inline images in the captions but the bug did not appear since they always have a explicitly set `width`.

I also ensured that inline images within captions are actually inlined (on the editor side this was already done). Technically this is a bug from Twentytwenty theme because it sets `img { display: block }` and doesnt override it for images within captions. But I don't see why this couldnt be a default set in Gutenberg?

One question is, does this break any legacy image markup from the early days of Gutenberg?

## How has this been tested?

Tested by enabling JetPack, adding a wide aligned image block and using `$latex d_f$` as a caption.

## Screenshots

Screen Shot 2020-06-20 at 14 46 25

Screen Shot 2020-06-20 at 14 46 56

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/image/editor.scss [hunk 1]
```diff
@@ -9,10 +9,6 @@ figure.wp-block-image:not(.wp-block) {
 		opacity: 0.3;
 	}
 
-	figcaption img {
```
Comment (jasmussen) [inline packages/block-library/src/image/editor.scss:12]: Nice, the more we can move from editor.scss to style.scss, the better shape we are in.

File: packages/block-library/src/image/style.scss [hunk 2]
```diff
@@ -11,8 +11,8 @@
 		text-align: center;
 	}
 
-	&.alignfull img,
-	&.alignwide img {
+	&.alignfull > img,
```
Comment (jasmussen) [inline packages/block-library/src/image/style.scss:14]: I suppose it's better to add specificity to here to be specific about the inheritance, than accidentally override the width on an image in a caption.

**Review by jasmussen — APPROVED** [review]
Thanks for your code cleanups and fixes. This tests well for me:

Screenshot 2020-06-22 at 08 34 23
Screenshot 2020-06-22 at 08 34 32
Screenshot 2020-06-22 at 08 35 16
Screenshot 2020-06-22 at 08 35 24

Don't mind the non-centered caption on the frontend, that's my theme's fault. 

Nice!

File: packages/block-library/src/image/style.scss (hunk 2 above)
Comment (oxyc) [reply packages/block-library/src/image/style.scss:14]: The selector specificity remains the same though (0,2,1), so a theme using `.wp-block-image.alignfull img´ will still override this like it used to.

Comment (jasmussen) [reply packages/block-library/src/image/style.scss:14]: How do you calculate specificity? This is not something I've learned yet, eager to glean a trick!

Comment (oxyc) [reply packages/block-library/src/image/style.scss:14]: Playing with https://specificity.keegan.st/ explains it by testing pretty well. Excluding inline styles and `!important`:

1. IDs
2. Classes, attributes, pseudo classes (:hover)
3. Elements and pseudo elements (:before, :after)

So `#id .wp-block-image.is-large img` is (1,2,1). To override it you need to either match or override something on the left side. Eg these would override:

- `#id .wp-block-image.is-large img` (1,2,1)
- `#id#id img` (2,0,1)

These would not:

- `.wp-block-image.wp-block-image.wp-block-image.wp-block-image.is-large img` (0,5,1)

And `div img` (0,0,2) would be overridden by eg `[src]` (0,1,0)

Comment (jasmussen) [reply packages/block-library/src/image/style.scss:14]: Nice. Thank you!

**Comment by Mamaduka** [comment]
Hello, @oxyc

It looks like the original issue got closed. Are you still able to reproduce the problem?

**Comment by Thelmachido** [comment]
I’m going to go ahead and close out this issue since there hasn’t been any activity on it since 2021. @oxyc feel free to reopen if you want to circle back on this.

**Pull request closed without merging**